Lint Error Android Build

<!--
Hi there and thank you for reporting a bug! 🐛🐛🐛

If you want to submit a feature request, use this link instead:
  https://github.com/rebeccahughes/react-native-device-info/issues/new?template=feature_request.md
-->

## Summary

when running :react-native-device-info:lint I get an error. Originally it was complaining about RNDeviceModule.java overriding or implementing a deprecated API, but I can't reproduce that error. I have attached the lint results print out.

|             |     |
| ----------- | --- |
| Version     | 0.15.2  |
| Affected OS | ?   |
| OS Version  | ?   |

## Current behavior

<!-- Describe the issue you are facing, including any available error
message, logs, stack trace .. -->

## Expected behavior

<!-- What should have happened instead of this bug -->

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-device-info:lint'.
> Lint found errors in the project; aborting build.
  
  Fix the issues identified by lint, or add the following to your build script to proceed with errors:
  ...
  android {
      lintOptions {
          abortOnError false
      }
  }
  ...
  
  Errors found:
  
  /home/stanimir/structmedia/triplogapp/TripApp/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java:64: Error: Missing permissions required by WifiManager.getConnectionInfo: android.permission.ACCESS_WIFI_STATE [MissingPermission]
        this.wifiInfo = manager.getConnectionInfo();
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/stanimir/structmedia/triplogapp/TripApp/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java:295: Error: Missing permissions required by TelephonyManager.getLine1Number: android.permission.READ_PHONE_STATE or android.permission.READ_SMS or android.permission.READ_PHONE_NUMBERS [MissingPermission]
        constants.put("phoneNumber", telMgr.getLine1Number());
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/stanimir/structmedia/triplogapp/TripApp/node_modules/react-native-device-info/android/build.gradle: Error: All com.android.support libraries must use the exact same version specification (mixing versions can lead to runtime crashes). Found versions 25.2.0, 23.0.1. Examples include com.android.support:support-compat:25.2.0 and com.android.support:appcompat-v7:23.0.1 [GradleCompatible]

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 ## Release Notes
 
+### 0.29.1
+ * Fix lint errors (library dependences, ignore permissions) (https://github.com/react-native-community/react-native-device-info/pull/590)
+
 ### 0.29.0
  * Add `isAutoDateAndTime()` and `isAutoTimeZone()` (https://github.com/rebeccahughes/react-native-device-info/pull/583)
 

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -4,6 +4,7 @@ def DEFAULT_COMPILE_SDK_VERSION             = 28
 def DEFAULT_BUILD_TOOLS_VERSION             = "28.0.0"
 def DEFAULT_TARGET_SDK_VERSION              = 28
 def DEFAULT_GOOGLE_PLAY_SERVICES_VERSION    = "+"
+def DEFAULT_SUPPORT_LIB_VERSION             = "28.0.0"
 
 android {
     compileSdkVersion project.hasProperty('compileSdkVersion') ? project.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
@@ -19,13 +20,17 @@ android {
         }
     }
     lintOptions {
-       warning 'InvalidPackage'
+       warning 'InvalidPackage', 'MissingPermission'
     }
 }
 
 dependencies {
     def googlePlayServicesVersion = project.hasProperty('googlePlayServicesVersion') ? project.googlePlayServicesVersion : DEFAULT_GOOGLE_PLAY_SERVICES_VERSION
+    def supportLibVersion = project.hasProperty('supportLibVersion') ? project.supportLibVersion : DEFAULT_SUPPORT_LIB_VERSION
 
     implementation 'com.facebook.react:react-native:+'
     implementation "com.google.android.gms:play-services-gcm:$googlePlayServicesVersion"
+    implementation "com.android.support:appcompat-v7:$supportLibVersion"
+    implementation "com.android.support:support-v4:$supportLibVersion"
+    implementation "com.android.support:support-media-compat:$supportLibVersion"
 }

